expression/expression.go

```diff
@@ -979,7 +979,7 @@ func scalarExprSupportedByTiKV(sf *ScalarFunction) bool {
 		ast.Length, ast.BitLength, ast.Concat, ast.ConcatWS, ast.Replace, ast.ASCII, ast.Hex,
 		ast.Reverse, ast.LTrim, ast.RTrim, ast.Strcmp, ast.Space, ast.Elt, ast.Field,
 		InternalFuncFromBinary, InternalFuncToBinary, ast.Mid, ast.Substring, ast.Substr, ast.CharLength,
-		ast.Right, /* ast.Left */
+		ast.Right /* ast.Left */, ast.RegexpLike, ast.RegexpSubstr, ast.RegexpInStr, ast.RegexpReplace,
```

Comment:
1. Regexp should also be pushed to TiKV
2. As discussed offline, for `Regexp/RegexpLike`, do not push down to TiKV if the collation is binary
3. Should add some tests